scripts/dss_deployer.py

```diff
@@ -0,0 +1,403 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import time
+import json
+import shutil
+import argparse
+import subprocess
+from collections import OrderedDict
+
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+
+
+parser = argparse.ArgumentParser(description='Create or Destroy a DSS Deployment')
+parser.add_argument('action', type=str, choices=['deploy', 'destroy'])
+parser.add_argument('--no-test-buckets', dest='create_test_buckets', action='store_false', default=True)
+parser.add_argument('--accept-defaults', dest='accept_defaults', action='store_true', default=False)
+parser.add_argument('--virtualenv', dest='virtualenv', default=os.path.join(pkg_root, 'virtualenv'))
+
+
+def no_action(msg):
+    print(f'NO ACTION {msg}')
+
+
+def find(cmd):
+    path = shutil.which(cmd)
+
+    if not os.path.isfile(path):
+        raise Exception(f'Command not fond: {cmd}')
+
+    return path
+
+
+def request_user_input(key, default=None):
+    if default is None:
+        return input(f'{key.upper()}=')
+    else:
+        if args.accept_defaults:
+            print(f'setting {key.upper()}={default}')
+            return default
+        return input(f'{key.upper()}[default={default}]=') or default
+
+
+def request_user_continuation(msg, default=True):
+    print(msg)
+
+    if args.accept_defaults:
+        return
+
+    if default:
+        resp = input('Proceed (Y/n)').lower() or 'y'
+    else:
+        resp = input('Proceed (y/N)').lower() or 'n'
+
+    if 'y' != resp:
+        sys.exit(1)
```

Comment:
Instead of writing this yourself, consider using http://click.pocoo.org/5/prompts/